wcコマンドを実装しました。

sht3ng opened the pull request.

sht3ng committed:
wc.rbファイルを作成しました。

sht3ng committed:
wcコマンドを実装しました。

メソッドの組み合わせで実装しています。

sht3ng committed:
-l付きのlsコマンドで表示される最後のファイル名の後に空白が表示される問題を修正しました。

sht3ng committed:
集計行の追加方法を変更しました。

集計行を追加するかどうかの判定をadd_total_rowメソッドの中で行っていたのですが、外に出しました。
add_total_rowの処理をwc_tablesに集計行を追加するのではなく、集計行のハッシュを戻すように変更しました。
add_total_rowメソッドの名前をcalculate_totalsに変更しました。

maedana commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,116 @@
+    total_bytesize += wc_table[:bytesize]
+  end
+  total_bytesize
+end
```
calculate_total_lines, calculate_total_words, calculate_total_bytesizeの3つは`wc_table`のキーが違うだけでやってること同じですかね? だとしたら共通化したいかなと思いました。

あとnil判定いるんですかね? もし必要だとしても、
```ruby
wc_tables.map {|wt| wt[:bytesize]}.compact.sum
```
とかだとスッキリしそうですね。

sht3ng replied:
ご助言いただいた方法だとかなりスッキリしますね！
教えてくださりありがとうございます😄

maedana commented on 06.wc/wc.rb:
```diff
@@ -0,0 +1,116 @@
+  return wc_from_stdin if paths.empty?
+
+  wc_tables = generate_wc_tables(paths)
+  output_wc(wc_tables, option_l: option_l)
```
`wc_from_stdin`と`output_wc`は標準入力からの処理とファイルからの処理で別れているのかなと思いました。命名としては対になってるほうが読みやすそうに感じますがいかがでしょう?

あと`wc_from_stdin`は内部で出力処理もやってますし、`output`はつけたほうが意図が伝わるかなと。`wc_from_stdin`という命名にするなら出力処理は呼び出し元でやりたい感じがします。そうなると、`wc`というメソッド名も出力のニュアンスないので付けたほうがわかりやすそうですね。

sht3ng replied:
確かにメソッド名に統一性がなかったです！
アドバイスくださりありがとうございます🙇‍♂️

sht3ng committed:
合計計算の処理方法を変更しました。

`wc_tables`のキーも合計計算メソッドに渡すように変更して合計計算メソッドを1つにしました。
合計の計算方法をご提案いただいた一行で済む簡潔な書き方に変更しました。
これに伴って、メソッド名を変更しました。
calculate_total_lines, calculate_total_rows, calculate_total_bytesize =>
calculate_total
calculate_totals => generate_total_row

sht3ng committed:
メソッド名の修正を行いました。

wc => output_wc
wc_from_stdin => output_wc_of_stdin
output_wc => output_wc_of_files
